language/go/package.go

```diff
@@ -34,6 +34,7 @@ import (
 type goPackage struct {
 	name, dir, rel        string
 	library, binary, test goTarget
+  tests                 []goTarget
 	proto                 protoTarget
 	hasTestdata           bool
```

Comment:
Oops sorry didn't see that this was already merged. I tested and this works great, I think the `maintest.go` issues is safe because this feature is guarded by a directive anyway. Nice work!